ipa-core/src/protocol/ipa_prf/malicious_security/hashing.rs

```diff
@@ -53,10 +53,16 @@ where
 /// This function takes two hash a vector of field elements into a single field element
 /// # Panics
 /// does not panic
```

Comment:
We should probably add a comment explaining the meaning of the `exclude_to` param...

```
The random challenge point `r` must be uniformly drawn from the field `F`, with the constraint that it does NOT appear in the set {0, 1, ..., λ-1}. 
In the paper `https://eprint.iacr.org/2023/909.pdf` this is shown on step (2.f) of protocol C.2.1 on page 21 where it says:
"The parties call `F_{coin}` to receive a random `r ∈ F_{p} \ [λ]`"

When using Fiat-Shamir to generate the random challenge point, we simply constrain the conversion from SHA-derived-entropy to field element to only generate valid values, rather than use rejection sampling.